runtime/jcl/common/jcldefine.c

```diff
@@ -131,8 +131,14 @@ defineClassCommon(JNIEnv *env, jobject classLoaderObject,
 	if (vmFuncs->hashClassTableAt(classLoader, utf8Name, utf8Length) != NULL) {
 		/* Bad, we have already defined this class - fail */
 		omrthread_monitor_exit(vm->classTableMutex);
+
 		if (J9_ARE_NO_BITS_SET(options, J9_FINDCLASS_FLAG_NAME_IS_INVALID)) {
-			vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGLINKAGEERROR, (UDATA *)*(j9object_t*)className);
```

Comment:
were you hitting this exception on warm runs?

Reply:
yeah without this

Reply:
this was the source of the linkage error initially

Reply:
Im not completely sure but it looks like Main Class loading is done via jcldefine

Reply:
Okay, we'll need to think about how to tell the difference between a valid classLoad and a multiple definition case.

It doesn't have to be solved now, but a `TODO: ` comment should be added so that it is not forgotten.

Reply:
is it not supposed to hit jcldefine. I noticed on cold runs there are three times it calls loadNonArrayClass. The first two times backtrace shows 
![image](https://user-images.githubusercontent.com/50454737/61658535-8864fb00-ac93-11e9-8e46-e5b893cd5344.png)

with two different classloaders (App and Extension) but the thing is they pass in `J9_FINDCLASS_FLAG_EXISTING_ONLY` in the options so it doesn't create the class. Afterwards when it hits it the backtrace shows jcldefine and works properly. For the warm run I notice it only hits loadNonArrayClass for Driver once using the extension class loader and never the application class loader